### PR TITLE
Python version check in ah_bootstrap

### LINKED ISSUE #380: ah_bootstrap.py should give a better message about non-Python 3 support
Apologies if this has already come up elsewhere, though I didn't see a ticket for it.  Installing Astropy 3.0 in Python 2 resulted in:

```
$ pip install astropy
Collecting astropy
  Downloading astropy-3.0.tar.gz (8.0MB)
    100% |████████████████████████████████| 8.0MB 134kB/s
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-m82yXL/astropy/setup.py", line 8, in <module>
        import ah_bootstrap
      File "ah_bootstrap.py", line 923, in <module>
        BOOTSTRAPPER = _Bootstrapper.main()
      File "ah_bootstrap.py", line 227, in main
        bootstrapper.run()
      File "ah_bootstrap.py", line 325, in run
        dist = method()
      File "ah_bootstrap.py", line 393, in get_local_directory_dist
        upgrade = self._do_upgrade(dist)
      File "ah_bootstrap.py", line 572, in _do_upgrade
        return self._do_download(version=upgrade.version)
      File "ah_bootstrap.py", line 557, in _do_download
        raise Exception(msg.format(DIST_NAME, source, repr(e)))
    Exception: Error retrieving astropy-helpers from PyPI:
    ImportError('No module named builtins',)
```

It would be nice if there were a `try/except` around this initial import, and upon failure an error message specifically that Astropy 3.0 doesn't support Python 2.

### PR DESCRIPTION
**Pull request by pllim**
Fix #380 

Companion of astropy/astropy#7316

c/c @astrofrog @bsipocz @drdavella

### CODE REVIEW

File: ah_bootstrap.py [hunk 1]
```diff
@@ -38,14 +38,17 @@
 
 import contextlib
 import errno
-import imp
```
Comment (pllim) [inline ah_bootstrap.py]: Bonus: Remove unused import.

**Comment by bsipocz** [comment]
I'm milestoning this to 2.0.7, but will need to be careful when backporting to update the version to be 2.7 rather than 3.5...

**Comment by pllim** [comment]
@bsipocz , I *think* I addressed your comment. Please see if I misunderstood. Also, please let me know if you want me to open a separate PR directly against `v2.0.x` branch. Thank you!

**Comment by bsipocz** [comment]
@pllim - No need to open another PR, I'll take care of it during backport.

**Comment by bsipocz** [comment]
Thanks @pllim!

**Pull request merged**